packages/di/src/index.ts

```diff
@@ -38,21 +40,40 @@ export class DIService {
   }
 
   addService<T>(classType: T): void {
-    DIService._ServiceSet.add(classType);
-    const myClass = classType as unknown as new () => InstanceOf<T>;
+    this.registerSingleton(classType as unknown as new () => InstanceOf<T>);
```

Comment:
generics at this point don't really matter as we are casting for no other reason other than to get the compiler working

Reply:
WHat would be nice is if all classes implemented an interface, so we could just do

`T extends IDiscordClass` or something

Reply:
let's not force that, simple is cool.

Reply:
Just do `T extends new () => InstanceOf<T>`

Reply:
> Just do `T extends new () => InstanceOf<T>`

can't because this lead back to original class reference, which is `Record<string, any>` or `Function`. If you like to go in depth, let me know on discord, I will brief you.